### PR TITLE
Fix 1679

### PR DESCRIPTION
**Pull request by bgth**
Appended parent name to the area name so that it becomes unique across the graph.

### CODE REVIEW

File: src/models/stackedAreaChart.js [hunk 1]
```diff
@@ -412,7 +412,12 @@ nv.models.stackedAreaChart = function() {
                         pointIndex = nv.interactiveBisect(series.values, e.pointXValue, chart.x());
                         var point = series.values[pointIndex];
                         var pointYValue = chart.y()(point, pointIndex);
-                        if (pointYValue != null) {
+                        if (pointYValue != null && pointYValue > 0) {
+                            stacked.highlightPoint(i, pointIndex, true);
+                        }
+                    
+                        // Draw at least one point if all values are zero.
+                        if (i === (data.length - 1) && !atleastOnePoint) {
```
Comment (liquidpele) [inline src/models/stackedAreaChart.js]: This change looks unrelated to the PR title?

Comment (bgth) [inline src/models/stackedAreaChart.js]: This one is for 1879

Comment (bgth) [inline src/models/stackedAreaChart.js]: Sorry 1876

Comment (bgth) [inline src/models/stackedAreaChart.js]: @liquidpele I tried porting the d3 v4 into the present nvd3 code using separate branch. I am facing issues. d3 v4 has changed alot from d3 v3. I was trying to only build lineChart.js and commented out all the other tests. It fails at the noData test itself as in d3 v4, the d3.select does not return undefined, it returns a selection object. I have commited the changes to testBranch - https://github.com/bgth/nvd3/tree/testBranch. Or may be my analysis is wrong. Is it possible for you to check? Thanks.

**Comment by liquidpele** [comment]
Okay, seems reasonable.

**Comment by bgth** [comment]
is there anything pending in this pull request? Can we close this if not?

**Comment by liquidpele** [comment]
This PR is merged already.
